app/api/lectures.js

```diff
@@ -0,0 +1,276 @@
+            )
+            full_response['questions'] = questions_in_lec
+        }
+        catch (e) {
+            res.status(400).send({error: "Unable to get lecture or questions"})
+            return
+        }
+        res.status(200).send(full_response)
+    }
+    else {  // if student, only send published info
+        let sectionLectureRelation = await getSectionLectureRelation(lectureId, enrollment.sectionId)
+        if (sectionLectureRelation == null) {
+            res.status(404).send({error: "This lecture does not exist in your section"})
+        }
+        else if (!sectionLectureRelation.published) {    // if this lecture (from user's section) isn't published
+            res.status(404).send({error: "This lecture is not yet published"})
+        }
+        else {  // if this lecture (from user's section) is published
+            try {
+                const lecture = await getLecture(lectureId)
+                full_response['lecture'] = lecture
+    
+                const questions_in_lec = await db.sequelize.query(  // raw sql query to get all published questions in this lecture using `QuestionInLecture`
+                    `SELECT q.* FROM Questions q INNER JOIN QuestionInLectures ql ON q.id = ql.questionId INNER JOIN Lectures l ON ql.lectureId = l.id WHERE ql.published = 1 AND l.id = ${lectureId}`
+                )
+                full_response['questions'] = questions_in_lec[0]    // index 0, because query above returns 2 duplicate instances of result
+            }
+            catch (e) {
+                res.status(400).send({error: "Unable to get lecture or questions"})
+                return
+            }
+            res.status(200).json(full_response)   // all good, return updated lecture object
+        }
+    }
+})
+
+// delete lecture and ALL relations to this lecture
+router.delete('/:lecture_id', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub)
+    const lectureId = req.params.lecture_id
+    const courseId = req.params.course_id
+    const enrollment = await getEnrollmentFromCourse(user.id, courseId) || await getEnrollmentFromSectionInCourse(user.id, courseId)
+    const lecture = await getLecture(lectureId)
+    
+    if (lecture == null) {   // if passed in lectureId does not exist
+        res.status(204).send({error: "This lecture does not exist already"})
+        return
+    }
+    else if (enrollment == null) {   // if user is not enrolled in this course
+        res.status(403).send()
+        return
+    }
+    else if (enrollment.role != 'teacher') {
+        res.status(403).send()
+        return
+    }
+    else {  // if user is a teacher
+        try {
+            await db.Lecture.destroy({  // delete from lecture table
+                where: {
+                    id: lectureId
+                }
+            })
+            await db.QuestionInLecture.destroy({    // delete from lecture-question relationship
```

Comment:
I think this will CASCADE DELETE when Lecture is deleted - it should say in the model or migration if so